File: app/tank-web/platform/tauri/client/trash.ts
import { invoke } from '@tauri-apps/api/core';
import type { TrashedMemo } from '@/types/trash';

export const trash = {
  list: () => invoke<TrashedMemo[]>('list_trashed_memos'),
  restore: (id: string) => invoke<boolean>('restore_trashed_memo', { id }),
  deleteForever: (id: string) =>
    invoke<boolean>('permanently_delete_trashed_memo', { id }),
  empty: () => invoke<boolean>('empty_trash'),
};

File: app/tank-desktop/src/commands/memo/trash.rs
// ==================== Trash / Recycle Bin ====================

use tauri::State;

use tank_core::MemoService;

use crate::app::state::AppState;
use crate::lock_utils::read_lock;

#[tauri::command]
pub fn list_trashed_memos(state: State<AppState>) -> Vec<tank_core::memo_file::TrashedMemo> {
    MemoService::new(&read_lock(&state.memo_file, "memo_file"))
        .list_trashed_memos()
        .unwrap_or_default()
}

#[tauri::command]
pub fn restore_trashed_memo(id: String, state: State<AppState>) -> bool {
    MemoService::new(&read_lock(&state.memo_file, "memo_file"))
        .restore_trashed_memo(&id)
        .unwrap_or(false)
}

#[tauri::command]
pub fn permanently_delete_trashed_memo(id: String, state: State<AppState>) -> bool {
    MemoService::new(&read_lock(&state.memo_file, "memo_file"))
        .permanently_delete_trashed_memo(&id)
        .unwrap_or(false)
}

#[tauri::command]
pub fn empty_trash(state: State<AppState>) -> bool {
    MemoService::new(&read_lock(&state.memo_file, "memo_file"))
        .empty_trash()
        .is_ok()
}
